includes/admin/wanderlist-admin.php

```diff
@@ -45,19 +61,34 @@ function wanderlist_geolocation_box() {
-
-	// Message field to show what's going on behind-the-scenes
-	echo '<div id="wanderlist-geocoder-message" class="wanderlist-message">Your location has been set to <strong class="place"></strong>.</div>';
-
-	// Hidden fields into which we can input data returned from our geocoder
-	echo '<input id="wanderlist-city" name="wanderlist-city" type="hidden" value="' . esc_attr( get_post_meta( $post->ID, 'wanderlist-city', true ) ) . '" />';
-	echo '<input id="wanderlist-region" name="wanderlist-region" type="hidden" value="' . esc_attr( get_post_meta( $post->ID, 'wanderlist-region', true ) ) . '" />';
-	echo '<input id="wanderlist-lng" name="wanderlist-lng" type="hidden" value="' . esc_attr( get_post_meta( $post->ID, 'wanderlist-lng', true ) ) . '" />';
-	echo '<input id="wanderlist-lat" name="wanderlist-lat" type="hidden" value="' . esc_attr( get_post_meta( $post->ID, 'wanderlist-lat', true ) ) . '" />';
-
-	echo '<input id="wanderlist-country" name="wanderlist-country" type="hidden" value="' . esc_attr( ltrim( wanderlist_place_data( 'country' ), ', ' ) ) . '"/>';
+	$options = get_option( 'wanderlist_settings' ); ?>
+
+	<input id="wanderlist-geolocation-input" data-mapboxkey="<?php esc_attr_e( $options['wanderlist_mapbox_key'] ); ?>" type="hidden" name="wanderlist-geolocation">
+	<div id="wanderlist-geolocation-map">
+		<a href="#" id="wanderlist-locate-user"><span class="screen-reader-text">Find me</span></a>
+		<div class="wanderlist-loader">
+  		<span class="wanderlist-loader-circle"></span>
+  		<span class="wanderlist-loader-circle"></span>
+		</div>
+	</div>
+
+	<?php // Message field to show what's going on behind-the-scenes ?>
+	<div id="wanderlist-geocoder-message" class="wanderlist-message">
+		<?php
+		printf( '<span class="success-message">%s <strong class="place"></strong>.</span>' ,
+				esc_html__( 'Your location has been set to', 'wanderlist' )
+		);
+		?>
+		<span class="error-message"></span>
+	</div>
+
+	<?php // Hidden fields into which we can input data returned from our geocoder. ?>
+	<input id="wanderlist-city" name="wanderlist-city" type="hidden" value="<?php esc_attr_e( get_post_meta( $post->ID, 'wanderlist-city', true ) ); ?>" />
+	<input id="wanderlist-region" name="wanderlist-region" type="hidden" value="<?php esc_attr_e( get_post_meta( $post->ID, 'wanderlist-region', true ) ); ?>" />
+	<input id="wanderlist-lng" name="wanderlist-lng" type="hidden" value="<?php esc_attr_e( get_post_meta( $post->ID, 'wanderlist-lng', true  ) ); ?>" />
+	<input id="wanderlist-lat" name="wanderlist-lat" type="hidden" value="<?php esc_attr_e( get_post_meta( $post->ID, 'wanderlist-lat', true  ) ); ?>" />
+	<input id="wanderlist-country" name="wanderlist-country" type="hidden" value="<?php esc_attr_e( ltrim( wanderlist_place_data( 'country' ), ', '  ) ); ?>" />
```

Comment:
Cool, makes sense.

- Matthew Riley MacPherson (Sent from mobile)

> On 26 Dec 2015, at 20:07, sarah ✈ semark <notifications@github.com> wrote:
> 
> In includes/admin/wanderlist-admin.php:
> 
> > +	<?php // Message field to show what's going on behind-the-scenes ?>
> > +	<div id="wanderlist-geocoder-message" class="wanderlist-message">
> > +		<?php
> > +		printf( '<span class="success-message">%s <strong class="place"></strong>.</span>' ,
> > +				esc_html__( 'Your location has been set to', 'wanderlist' )
> > +		);
> > +		?>
> > +		<span class="error-message"></span>
> > +	</div>
> > +
> > +	<?php // Hidden fields into which we can input data returned from our geocoder. ?>
> > +	<input id="wanderlist-city" name="wanderlist-city" type="hidden" value="<?php esc_attr_e( get_post_meta( $post->ID, 'wanderlist-city', true ) ); ?>" />
> > +	<input id="wanderlist-region" name="wanderlist-region" type="hidden" value="<?php esc_attr_e( get_post_meta( $post->ID, 'wanderlist-region', true ) ); ?>" />
> > +	<input id="wanderlist-lng" name="wanderlist-lng" type="hidden" value="<?php esc_attr_e( get_post_meta( $post->ID, 'wanderlist-lng', true  ) ); ?>" />
> > +	<input id="wanderlist-lat" name="wanderlist-lat" type="hidden" value="<?php esc_attr_e( get_post_meta( $post->ID, 'wanderlist-lat', true  ) ); ?>" />
> > +	<input id="wanderlist-country" name="wanderlist-country" type="hidden" value="<?php esc_attr_e( ltrim( wanderlist_place_data( 'country' ), ', '  ) ); ?>" />
> get_post_meta is a core WordPress function; getting the country is a teensy bit more complicated, and since it's done in a few places, there's a reusable function (https://github.com/sarahmonster/wanderlist/blob/master/includes/wanderlist.php#L15) to access the data in a consistent manner.
> 
> I've reworked the function to output the data in a saner way, which cleans this up a bit.
> 
> —
> Reply to this email directly or view it on GitHub.
>